packages/react/src/ErrorBoundary.tsx

```diff
@@ -0,0 +1,52 @@
+import { Component, ErrorInfo, ReactElement, ReactNode, isValidElement } from 'react';
+import { BacktraceClient } from './BacktraceClient';
+
+type RenderFallback = () => ReactElement;
+
+export interface Props {
+    children: ReactNode;
+    fallback?: ReactElement | RenderFallback;
+}
+
+export interface State {
+    hasError: boolean;
+    error?: Error;
+}
+
+export class ErrorBoundary extends Component<Props, State> {
+    private _client: BacktraceClient;
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            hasError: false,
+            error: undefined,
+        };
+        // grabbing here so it will fail fast if BacktraceClient is uninitialized
+        this._client = BacktraceClient.instance;
+    }
+
+    static getDerivedStateFromError(error: Error) {
+        return { hasError: true, error };
+    }
+
+    componentDidCatch(error: Error, info: ErrorInfo) {
```

Comment:
shouldn't we also add accessors like public or private? We use it everywhere in the code. Is there any reason why we shouldn't continue doing that here?

Reply:
Good call I can add them

Reply:
Fixed in latest PR